core/database/foxx/tests/validation.test.js

```diff
@@ -0,0 +1,352 @@
+                admins: ["user1"],
+            };
+        }
+
+        it("should accept valid metadata-only configuration", function () {
+            const config = getValidMetadataConfig();
+            const result = validateMetadataConfig(config);
+            expect(result.ok).to.be.true;
+        });
+
+        it("should reject configuration with Globus fields", function () {
+            const config = getValidMetadataConfig();
+            config.pub_key = "ssh-rsa AAAAB3...";
+            config.endpoint = "endpoint-id";
+            const result = validateMetadataConfig(config);
+            expect(result.ok).to.be.false;
+            expect(result.error.message).to.include("should not have: pub_key, endpoint");
+        });
+
+        it("should reject configuration with path field", function () {
+            const config = getValidMetadataConfig();
+            config.path = "/data/repos/test-repo";
+            const result = validateMetadataConfig(config);
+            expect(result.ok).to.be.false;
+            expect(result.error.message).to.include("should not have: path");
+        });
+    });
+
+    describe("validateAllocationParams", function () {
+        it("should accept valid allocation parameters", function () {
+            const params = {
+                subject: "user123",
+                size: 1000000,
+            };
+            const result = validateAllocationParams(params);
+            expect(result.ok).to.be.true;
+        });
+
+        it("should accept allocation with path", function () {
+            const params = {
+                subject: "user123",
+                size: 1000000,
+                path: "/custom/path",
+            };
+            const result = validateAllocationParams(params);
+            expect(result.ok).to.be.true;
+        });
+
+        it("should reject missing subject", function () {
+            const params = {
+                size: 1000000,
+            };
+            const result = validateAllocationParams(params);
+            expect(result.ok).to.be.false;
+            expect(result.error.message).to.include("Allocation subject is required");
+        });
+
+        it("should reject zero or negative size", function () {
+            const params = {
+                subject: "user123",
+                size: 0,
+            };
+            const result = validateAllocationParams(params);
+            expect(result.ok).to.be.false;
+            expect(result.error.message).to.include("Allocation size must be a positive number");
+        });
+
+        it("should reject non-string path", function () {
+            const params = {
+                subject: "user123",
+                size: 1000000,
+                path: 123,
+            };
+            const result = validateAllocationParams(params);
+            expect(result.ok).to.be.false;
+            expect(result.error.message).to.include("Allocation path must be a string");
```

Comment:
Woud be nice if it printed the actual type in the error message. 

"Allocation path must be a string but a number was provided"